### ISSUE TITLE
Error when closure on nested expression

### ISSUE THREAD

**Issue by Razican** [rust-lang-nursery/rustfmt#1434]
This code gives an error when trying to format with `rustfmt`: 

```rust
fn main() {
    for _ in 0..100 {
        let prototype_id = PrototypeIdData::from_reader::<_, B>(&mut self.file_cursor).chain_err(|| {
                           format!("could not read prototype ID at offset {:#010x}",
                                   current_offset)
                       })?;
    }
}
```

The biggest problem is that even if I fix it by putting the `.error_chain` call in the next line, it moves it back to the position above and gives the error.

```
Rustfmt failed at /home/razican/workspace/super/dalvik/src/read.rs:203: line exceeded maximum length (maximum: 100, found: 105) (sorry)
```

**Comment by topecongiro** [comment]
This is fixed via #1438.
The problem is that rustfmt incorrectly takes `.chain_err` as `?` at the end of the expression and hence thinks that it will fit into a single line.
However, since the actual width of each expression is 10 and 1 respectively, the line exceeds maximum length.

**Comment by nrc** [comment]
Closed by https://github.com/rust-lang-nursery/rustfmt/pull/1438
